chore: bump glog to 0.7.1

aleksraiden opened the pull request:
Update google log to 0.7.0. Full release notes here: https://github.com/google/glog/releases/tag/v0.7.0 
Also, included update 0.7.1 (see here: https://github.com/google/glog/releases/tag/v0.7.1)

**Key features**

- C++14 is now the primary language standard
- google::EnableLogCleaner(unsigned) has been superseded by google::EnableLogCleaner(std::chrono::minutes)
- google::LogSeverity has been converted to an unscoped enum. This can result in compilation issues if implicit conversion between an int and google::LogSeverity was erroneously assumed.
- A lot of fixes and enhancements (see full changelog)

**Code changes**

At this update we need to fix our codebase to have full comparability with this and futures release of glog. Thanks @torwig for this fixes.

- Fix google::EnableLogCleaner (done)
- Fix linting and code checks (done)
- Fix google::LogSeverity (done)
- Fix a symbolizes a program counter (in progress)

aleksraiden committed:
Update kvrocks.yaml

Update check-typos to 1.18.2

aleksraiden committed:
Update sonar.yaml

Update actions/checkout to v4

aleksraiden committed:
Merge branch 'apache:unstable' into unstable-head

aleksraiden committed:
Merge branch 'apache:unstable' into unstable-head

aleksraiden committed:
Merge branch 'apache:unstable' into unstable-head

aleksraiden committed:
Update glog.cmake

torwig committed:
Glog 0.7.0 errors (#11)

* Fix build errors related to GLOG

* Format code

* Fix

aleksraiden commented:
Now we have unresolved issue at build kvrocks:

> [100%] Linking CXX executable kvrocks
> /usr/sbin/ld: CMakeFiles/kvrocks.dir/src/cli/main.cc.o: in function `SegvHandler':
> /__w/kvrocks/kvrocks/src/cli/signal_util.h:54:(.text.SegvHandler[SegvHandler]+0x266): undefined reference to `google::Symbolize(void*, char*, unsigned long)'
> collect2: error: ld returned 1 exit status

torwig commented:
`google::Symbolize` isn't exported anymore in 0.7.0 so we need to use something else for our purposes (or ask the maintainers of GLOG to export it again :) ).

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
update to 0.7.1

PragmaTwice committed:
Merge branch 'unstable' into update-glog-0.7.0

PragmaTwice commented on src/cli/main.cc:
```diff
@@ -103,15 +104,17 @@ static void InitGoogleLog(const Config *config) {
     FLAGS_log_dir = config->log_dir + "/";
     if (config->log_retention_days != -1) {
-      google::EnableLogCleaner(config->log_retention_days);
+      auto minutes =
+          std::chrono::duration_cast<std::chrono::minutes>(std::chrono::hours(24) * config->log_retention_days);
+      google::EnableLogCleaner(minutes);
```
```suggestion
      google::EnableLogCleaner(std::chrono::hours(24) * config->log_retention_days);
```

PragmaTwice commented on src/config/config.cc:
```diff
@@ -553,7 +553,9 @@ void Config::initFieldCallback() {
 
              if (log_retention_days != -1) {
-               google::EnableLogCleaner(log_retention_days);
+               auto minutes =
+                   std::chrono::duration_cast<std::chrono::minutes>(std::chrono::hours(24) * log_retention_days);
+               google::EnableLogCleaner(minutes);
```
```suggestion
               google::EnableLogCleaner(std::chrono::hours(24) * log_retention_days);
```

PragmaTwice committed:
Apply suggestions from code review

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden commented:
Hmm, I think (thanks, @torwig) that in failed gotest kvrocks not starting yet. Need to investigate this

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

aleksraiden committed:
Update config.cc

Awoid a compile warning: comparison of integer expressions of different signedness: ‘const int’ and ‘const size_t’

torwig committed:
Rewrite the test of the log cleaner (#15)

aleksraiden committed:
Merge branch 'apache:unstable' into update-glog-0.7.0

aleksraiden commented:
Great thanks to @torwig about rewrite a gotest for logger. Wuuf, now a longest PR ready to review

git-hulk commented:
@aleksraiden Great!

aleksraiden committed:
Merge branch 'unstable' into update-glog-0.7.0

PragmaTwice committed:
Merge branch 'unstable' into update-glog-0.7.0
